pkg/scheduler/framework/plugins/nodevolumelimits/csi.go

```diff
@@ -103,7 +106,7 @@ func (pl *CSILimits) Filter(ctx context.Context, _ *framework.CycleState, pod *v
 
 	attachedVolumes := make(map[string]string)
 	for _, existingPod := range nodeInfo.Pods {
-		if err := pl.filterAttachableVolumes(csiNode, existingPod.Pod.Spec.Volumes, existingPod.Pod.Namespace, attachedVolumes); err != nil {
+		if err := pl.filterAttachableVolumes(pod, csiNode, existingPod.Pod.Spec.Volumes, existingPod.Pod.Namespace, attachedVolumes); err != nil {
```

Comment:
I think you want this to be `existingPod`, since on line 157 you set `pvcName = pod.Name + "-" + vol.Name`, where in this case `pod` is the pod being scheduled and `vol` is each existing pod volumes

Reply:
Good catch. I wonder why it didn't get caught by the unit tests.

Before I change it, would it make sense to remove the separate `volumes` and `namespace` parameters from `filterAttachableVolumes` and instead work only with the pod fields? It makes the function less flexible but simpler overall. I was undecided when adding the `pod` parameter.

Reply:
It wasn't caught because none of the unit tests use an existing pod which has a generic ephemeral volume. Do I really need to add also tests for that? :sob: